salt/grains/mdata.py

```diff
@@ -63,6 +63,8 @@ def _user_mdata(mdata_list=None, mdata_get=None):
         return grains
 
     for mdata_grain in __salt__['cmd.run'](mdata_list, ignore_retcode=True).splitlines():
+        if mdata_grain.startswith("ERROR:"):
```

Comment:
Would it be a good idea to log here? Or should this data only be in the `mdata_grain` with logging elsewhere?

Reply:
We were only getting here in a one case that was due to the broken is_smartos_XXX functions in py3, and this was throwing exceptions. With this if in place if we ever get here we skip the invalid date. I don't think this needs to be logged, as the only way to get here is if is_smartos_XXX() is broken again, just having the grain missing is OK in that case.

In theory we could remove this one and the one below on L113 again and we'll have exceptions -> error log entries again.

Reply:
Added some 'warning' level locking indicating that the mdata grains may be incomplete or missing, but this should not cause an message of level error or higher as it did before when an exception was thrown.